src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat/TryConvertProjectConverterStep.cs

```diff
@@ -36,61 +31,26 @@ public class TryConvertProjectConverterStep : UpgradeStep
         {
-            _restorer = restorer ?? throw new ArgumentNullException(nameof(restorer));
             _runner = runner ?? throw new ArgumentNullException(nameof(runner));
         }
 
         protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token) => Task.FromResult(context?.CurrentProject is not null);
 
-        protected async override Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeContext context, CancellationToken token)
+        protected override Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeContext context, CancellationToken token)
         {
             if (context is null)
             {
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var project = context.CurrentProject.Required();
-            var components = await project.GetComponentsAsync(token).ConfigureAwait(false);
-            if (components.HasFlag(ProjectComponents.XamarinAndroid) || components.HasFlag(ProjectComponents.XamariniOS))
-            {
-                context.Properties.SetPropertyValue("componentFlag", components.ToString(), true);
-            }
-
-            var result = await RunTryConvertAsync(context, context.CurrentProject.Required(), token).ConfigureAwait(false);
-
-            await _restorer.RestorePackagesAsync(context, context.CurrentProject.Required(), token).ConfigureAwait(false);
-
-            return result;
+            return _runner.ApplyAsync(context, context.CurrentProject.Required(), token);
         }
 
-        private async Task<UpgradeStepApplyResult> RunTryConvertAsync(IUpgradeContext context, IProject project, CancellationToken token)
-        {
-            Logger.LogInformation($"Converting project file format with try-convert{VersionString}");
-
-            var result = await _runner.RunAsync(context, project, token).ConfigureAwait(false);
-
-            // Reload the workspace since an external process worked on and
-            // may have changed the workspace's project.
-            await context.ReloadWorkspaceAsync(token).ConfigureAwait(false);
-
-            if (!result)
-            {
-                Logger.LogCritical("Conversion with try-convert failed.");
-                return new UpgradeStepApplyResult(UpgradeStepStatus.Failed, $"Conversion with try-convert failed.");
-            }
-            else
-            {
-                Logger.LogInformation("Project file converted successfully! The project may require additional changes to build successfully against the new .NET target.");
-                return new UpgradeStepApplyResult(UpgradeStepStatus.Complete, "Project file converted successfully");
-            }
-        }
-
-        protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
+        protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
```

Comment:
Why is this no longer async?

Reply:
It is still async. However, because the method doesn't await anymore, it can just pass on the task from the method it calls